SPRD SIPA Ethernet: preserve the custom NAPI polling weight on Linux 6.6.

Use the device-aware random MAC helper to update the netdev address and
addr_assign_type together, rather than writing its now-const dev_addr field.

References:
https://github.com/torvalds/linux/blob/v6.6/include/linux/netdevice.h
https://github.com/torvalds/linux/blob/v6.6/include/linux/etherdevice.h

--- a/package/mtk/applications/5g-modem/quectel_SRPD_PCIE/src/sipa/sipa_eth.c
+++ b/package/mtk/applications/5g-modem/quectel_SRPD_PCIE/src/sipa/sipa_eth.c
@@ -744,10 +744,10 @@
 	netdev->irq = 0;
 	netdev->dma = 0;
 	netdev->features &= ~(NETIF_F_IP_CSUM | NETIF_F_IPV6_CSUM | NETIF_F_HW_CSUM);
-	random_ether_addr(netdev->dev_addr);
+	eth_hw_addr_random(netdev);
         netdev->sysfs_groups[0] = &sipa_eth_attribute_group;
 
-	netif_napi_add(netdev,
+	netif_napi_add_weight(netdev,
 		       &sipa_eth->napi,
 		       sipa_eth_rx_poll_handler,
 		       SIPA_ETH_NAPI_WEIGHT);
